### PR TITLE
use attributes on status objects instead of dictionary

### PR DESCRIPTION
**Pull request by pfps**

### CODE REVIEW

File: lib/logitech_receiver/status.py [hunk 1]
```diff
@@ -128,103 +94,51 @@ def __init__(self, device, changed_callback):
...
-    def set_battery_keys(self, battery):
-        if battery is not None:
-            level, nextLevel, status, voltage = battery
-            self.set_battery_info(level, nextLevel, status, voltage)
-        elif self.get(KEYS.BATTERY_STATUS, None) is not None:
-            self[KEYS.BATTERY_STATUS] = None
-            self[KEYS.BATTERY_CHARGING] = None
-            self[KEYS.BATTERY_VOLTAGE] = None
-            self.changed()
+            self.set_battery_info(battery if battery is not None else Battery(None, None, None, None))
```
Comment (MattHag) [inline lib/logitech_receiver/status.py:138]: Please split this in at least two lines

File: lib/logitech_receiver/status.py [hunk 2]
```diff
@@ -253,10 +165,6 @@ def changed(self, active=None, alert=ALERT.NONE, reason=None, push=False):
...
                     _settings.apply_all_settings(d)
 
-            else:
-                if was_active:  # don't clear status when devices go inactive
-                    pass
-
-        # if logger.isEnabledFor(logging.DEBUG):
-        #     logger.debug("device %d changed: active=%s %s", d.number, self._active, dict(self))
+        if logger.isEnabledFor(logging.DEBUG):
+            logger.debug("device %d changed: active=%s %s", d.number, self._active, self.battery)
```
Comment (MattHag) [inline lib/logitech_receiver/status.py:169]: f-string would be cool

File: lib/logitech_receiver/common.py [hunk 3]
```diff
@@ -546,18 +551,57 @@ def __getattr__(self, k):
...
-)
 
-
-def BATTERY_OK(status):
-    return status not in (BATTERY_STATUS.invalid_battery, BATTERY_STATUS.thermal_error)
+@dataclass
+class Battery:
+    """Information about the current state of a battery"""
+
+    level: Optional[NamedInt | int]
```
Comment (MattHag) [inline lib/logitech_receiver/common.py:559]: For Python 3.7 `Optional[Union[NamedInt, int]]` should do the trick

File: lib/logitech_receiver/common.py [hunk 4]
```diff
@@ -546,18 +551,57 @@ def __getattr__(self, k):
...
+    next_level: Optional[NamedInt | int]
```
Comment (MattHag) [inline lib/logitech_receiver/common.py:560]: same here

File: lib/logitech_receiver/common.py [hunk 5]
```diff
@@ -546,18 +551,57 @@ def __getattr__(self, k):
...
+        return self.status in (
+            Battery.STATUS.recharging,
+            Battery.STATUS.almost_full,
+            Battery.STATUS.full,
+            Battery.STATUS.slow_recharge,
+        )
+
+    def to_str(self):
+        if isinstance(self.level, NamedInt):
+            return _("Battery: %(level)s (%(status)s)") % {"level": _(self.level), "status": _(self.status)}
```
Comment (MattHag) [inline lib/logitech_receiver/common.py:603]: Much, much shorter with f-strings.

return _("Battery: {_(self.level)} ({_(self.status)})")

Comment (pfps) [reply lib/logitech_receiver/common.py:603]: There needs to be an f in there somewhere.  And then how does translation work on f-strings?

File: lib/logitech_receiver/status.py (hunk 2 above)
Comment (pfps) [reply lib/logitech_receiver/status.py:169]: probably not worth it

File: lib/logitech_receiver/common.py (hunk 3 above)
Comment (pfps) [reply lib/logitech_receiver/common.py:559]: done

File: lib/logitech_receiver/common.py (hunk 4 above)
Comment (pfps) [reply lib/logitech_receiver/common.py:560]: ditto

File: lib/logitech_receiver/status.py (hunk 1 above)
Comment (pfps) [reply lib/logitech_receiver/status.py:138]: not common, but works

**Pull request merged**